flight_company.rb

```diff
@@ -1,155 +1,63 @@
 # frozen_string_literal: true
 
-require 'date'
-require 'net/http'
-require 'uri'
-require 'json'
+require './flight_validator_service'
+require './flight_data_service'
 
 ARRIVAL_RESPONSE = %w[S s sim Sim SIM].freeze
-AIRPORTS = File.read('airports.json')
 
 origin_airport = ''
 destination_airport = ''
 departure_time = ''
 arrival_time = ''
 
-file_envs = File.read('.env').split(%r{=|\n})
-ENV = Hash[*file_envs]
-
 puts '==========================================='
 puts '   Projeto de Companhia Aérea - Rebase'
 
 while origin_airport.empty?
   puts "\n- Qual é o aeroporto de origem? [Formato: abreviado com 3 letras, por exemplo: GRU]"
   origin = gets.chomp
-  if origin.match?(/^[a-zA-Z]{3}$/)
-    AIRPORTS.include?(origin.upcase) ? (origin_airport = origin) : (puts '=> Aeroporto inválido.')
-  else
-    puts '=> Na resposta deve conter apenas 3 letras sem caracteres especiais.'
-  end
+  origin_airport = FlightValidatorService.execute(origin).valid_airport
 end
 
 while destination_airport.empty?
   puts "\n- Qual é o aeroporto de destino? [Formato: abreviado com 3 letras, por exemplo: GRU]"
   destination = gets.chomp
-  if !destination.match?(/^[a-zA-Z]{3}$/)
-    puts '=> Na resposta deve conter apenas 3 letras sem caracteres especiais.'
-  elsif destination.eql?(origin_airport)
-    puts '=> Aeroporto de destino não pode ser o mesmo de origem.'
-  elsif AIRPORTS.include?(destination.upcase)
-    destination_airport = destination
-  else
-    puts '=> Aeroporto inválido.'
-  end
+  destination_airport = FlightValidatorService.execute(destination).valid_airport(origin)
 end
```

Comment:
Existe grandes diferenças entre o fluxo de partida e o destino? Questiono isso, pois dependendo de como for talvez seja possível usar um método só para eles e passar como parâmetro as informações necessárias, o que você acha?

Reply:
:thinking:  Mas já está sendo um método só para validar o fluxo de partida e o destino, a diferença é que no fluxo de destino temos que ter a variável de partida.

Reply:
Ajustado em: https://github.com/Laryssa-lfa/flight-company/pull/3/commits/c0f4b573cb3f7a196bae6f1e6f196da6bf4d0364